src/Product.php

```diff
@@ -655,6 +656,10 @@ protected function processVariationIdentifiers(array $variation)
             $this->setField('ordernumber', array());
         }
 
+        if ($this->getField('main_variation_id') == $this->getDefaultEmptyValue() || $variation['isMain']) {
```

Comment:
This won't cover an edge case (affects in fact just one customer) which is when the main variation is inactive. I would propose that you check for `$variation['mainVariationId']` and use that as value for `main_variation_id`. That field should be available on every variation (including main).